.github/workflows/release.yaml

```diff
@@ -4,15 +4,68 @@ on:
 
 name: Release
 jobs:
-  pypi:
-    name: PyPI Release
+  test-release:
     runs-on: ubuntu-latest
     steps:
-      - uses: actions/checkout@v3
-      - name: PyPI Upload
-        uses: FeatureLabs/gh-action-pypi-upload@v2
-        env:
-          PYPI_USERNAME: ${{ secrets.PYPI_USERNAME }}
-          PYPI_PASSWORD: ${{ secrets.PYPI_PASSWORD }}
-          TEST_PYPI_USERNAME: ${{ secrets.TEST_PYPI_USERNAME }}
-          TEST_PYPI_PASSWORD: ${{ secrets.TEST_PYPI_PASSWORD }}
+    - name: Check out code
+      uses: actions/checkout@v3
+
+    - name: Set up Python
+      uses: actions/setup-python@v4
+      with:
+        python-version: '3.11'
+
+    - name: Install Poetry
+      uses: snok/install-poetry@v1
+
+    - name: Configure Poetry for TestPyPI
+      run: |
+        poetry config repositories.testpypi https://test.pypi.org/legacy/
+        poetry config http-basic.testpypi ${{ secrets.TEST_PYPI_USERNAME }} ${{ secrets.TEST_PYPI_PASSWORD }}
+
+    - name: Install dependencies
+      run: poetry install
+
+    - name: Build package
+      run: poetry build
+
+    - name: Publish package to TestPyPI
+      run: poetry publish --repository testpypi
+
+  pypi-release:
+    runs-on: ubuntu-latest
+    needs: test-release
+    if: github.event.inputs.publish_to_pypi == 'yes'
```

Comment:
This is supposed to only run if the input is set, but I don't think I setup the input correctly. I don't see the input option in the GitHub actions UI.

https://docs.github.com/en/actions/learn-github-actions/contexts#inputs-context